WordPress/src/main/java/org/wordpress/android/ui/jetpack/scan/details/ThreatDetailsFragment.kt

```diff
@@ -0,0 +1,59 @@
+package org.wordpress.android.ui.jetpack.scan.details
+
+import android.os.Bundle
+import android.view.View
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.ViewModelProvider
+import kotlinx.android.synthetic.main.threat_details_fragment.*
+import org.wordpress.android.R
+import org.wordpress.android.WordPress
+import org.wordpress.android.ui.jetpack.scan.ScanConstants
+import org.wordpress.android.ui.jetpack.scan.details.ThreatDetailsViewModel.UiState.Content
+import org.wordpress.android.ui.jetpack.scan.details.adapters.ThreatDetailsAdapter
+import org.wordpress.android.ui.utils.UiHelpers
+import org.wordpress.android.util.image.ImageManager
+import javax.inject.Inject
+
+class ThreatDetailsFragment : Fragment(R.layout.threat_details_fragment) {
+    @Inject lateinit var imageManager: ImageManager
+    @Inject lateinit var uiHelpers: UiHelpers
+    @Inject lateinit var viewModelFactory: ViewModelProvider.Factory
+    private lateinit var viewModel: ThreatDetailsViewModel
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        initDagger()
+        initAdapter()
+        initViewModel()
+    }
+
+    private fun initDagger() {
+        (requireActivity().application as WordPress).component()?.inject(this)
+    }
+
+    private fun initAdapter() {
+        recycler_view.adapter = ThreatDetailsAdapter(imageManager, uiHelpers)
+    }
+
+    private fun initViewModel() {
+        viewModel = ViewModelProvider(this, viewModelFactory).get(ThreatDetailsViewModel::class.java)
+        setupObservers()
+        val threatId = requireActivity().intent.getLongExtra(ScanConstants.ARG_THREAT_ID, 0)
```

Comment:
Idea (💡 ): Why not moving this `ARG_THREAT_ID` constant (in a `companion object`) within the fragment itself and remove the need to create a `ScanConstants` class altogether?

Reply:
I created `ScanConstants` for keeping `intent` `args` because a similar separate class [ReaderConstants](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderConstants.java#L35) was used to keep them for `Reader` classes. 

No personal preference on this so moved it in 81e165c.

Reply:
Thanks Ashita! 🌟 

I wasn't sure about the intention here and whether this is a pattern you follow elsewhere for some specific reason. It just felt too much to create an extra class instead of keeping this close to its usage.

Maybe I am missing something. Let's keep a note on it and discuss it for next time.